### PR TITLE
Nested geofences with individual actions and max altitude

### PR DESCRIPTION
**Pull request by kibidev**
This PR continues on #22.

Needs https://github.com/aviant-tech/PX4-Autopilot/pull/50

### CODE REVIEW

**Review by Torjusba — APPROVED** [review]
Looks good! I left some suggestions and a question

File: src/MissionManager/PlanMasterController.cc [hunk 1]
```diff
@@ -345,9 +340,16 @@ void PlanMasterController::sendToVehicle(void)
     } else if (syncInProgress()) {
         qCWarning(PlanMasterControllerLog) << "PlanMasterController::sendToVehicle called while syncInProgress";
     } else {
-        qCDebug(PlanMasterControllerLog) << "PlanMasterController::sendToVehicle start mission sendToVehicle";
-        _sendGeoFence = true;
-        _missionController.sendToVehicle();
+        if (_geoFenceController.supported()) {
+            qCDebug(PlanMasterControllerLog) << "PlanMasterController::sendToVehicle start GeoFence sendToVehicle";
+            _geoFenceController.sendToVehicle();
+            _sendMission = true;
```
[the remark this answers is not in the document]
Comment (kibidev) [reply src/MissionManager/PlanMasterController.cc:346]: Just a coincidence, will change it to make it more consistent.

File: src/PlanView/GeoFenceEditor.qml [hunk 2]
```diff
@@ -199,17 +199,62 @@ QGCFlickable {
                         }
 
                         QGCLabel {
-                            text:               qsTr("Delete")
+                            text:               qsTr("Altitude")
```
[the remark this answers is not in the document]
Comment (kibidev) [reply src/PlanView/GeoFenceEditor.qml]: Fixed

**Comment by kibidev** [comment]
@Torjusba during fixing of altitude label to include "AMSL" I found some issues with the altitude field as well, that is addressed in a newly added commit.

File: src/PlanView/GeoFenceEditor.qml [hunk 3]
```diff
@@ -199,17 +200,70 @@ QGCFlickable {
...
 
+                        Repeater {
+                            model: myGeoFenceController.polygons
+
+                            QGCTextField {
+                                text: object.maxAltitude == 0 ? "Disabled" : object.maxAltitude
+                                textColor: acceptableInput ? "black" : "red"
+                                unitsLabel: "m"
+                                showUnits: true
+                                validator: RegExpValidator{ regExp: /^[0-9]{1,5}|disabled$/i }
```
Comment (Torjusba) [inline src/PlanView/GeoFenceEditor.qml:216]: According to https://doc.qt.io/qt-5/qml-qtquick-regexpvalidator.html#details, `RegExpValidator` is deprecated and should be replaced by `RegularExpressionValidator`

File: src/PlanView/GeoFenceEditor.qml [hunk 4]
```diff
@@ -285,17 +339,69 @@ QGCFlickable {
...
 
+                        Repeater {
+                            model: myGeoFenceController.circles
+
+                            QGCTextField {
+                                text: object.maxAltitude == 0 ? "Disabled" : object.maxAltitude
+                                textColor: acceptableInput ? "black" : "red"
+                                unitsLabel: "m"
+                                showUnits: true
+                                validator: RegExpValidator{ regExp: /^[0-9]{1,5}|disabled$/i }
```
Comment (Torjusba) [inline src/PlanView/GeoFenceEditor.qml:362]: See other comment on deprecated `RegExpValidator`

**Review by Torjusba — APPROVED** [review]
Looks good to me. I left some comments about `RegExpValidator` being deprecated, you can deal with it if you want to.

File: src/PlanView/GeoFenceEditor.qml (hunk 3 above)
Comment (kibidev) [reply src/PlanView/GeoFenceEditor.qml:216]: I think RegularExpressionValidator is not yet supported by QTextField as of qt 5.15, but will require update to some version of 6.x first.

**Pull request merged**